File: Problem2.js
let box = [];
let a = 0;
let b = 1;
let c = 1;
do {
	c = a + b;
  a = b;
  b = c;
  box.push(c);
} while (a + b < 4000000);

let even = function(num){
 return num % 2 ===0};

let sum = box.filter(even).reduce((total,amount)=> total + amount);

 console.log(sum);

File: Ruby/Problem2.rb
box = []
a = 0
b = 1
c = 1
while a + b < 4000000
  c = a + b;
  a = b;
  b = c;
  box.push(c);
end
box.delete_if {|n| n%2 != 0}

puts box.inject {|sum, n| sum + n}
